addons/calendar/models/calendar.py

```diff
@@ -1229,46 +1235,131 @@ def get_display_time_tz(self, tz=False):
                 rrule='',
                 recurrency=False,
-                final_date=datetime.strptime(data.get('start'), DEFAULT_SERVER_DATETIME_FORMAT if data['allday'] else DEFAULT_SERVER_DATETIME_FORMAT) + timedelta(hours=values.get('duration', False) or data.get('duration'))
+                final_date=datetime.strptime(meeting_origin_vals.get('start'), DEFAULT_SERVER_DATETIME_FORMAT if meeting_origin_vals['allday'] else DEFAULT_SERVER_DATETIME_FORMAT) + timedelta(hours=values.get('duration', False) or meeting_origin_vals.get('duration'))
             )
 
             # do not copy the id
-            if data.get('id'):
-                del data['id']
-            return meeting_origin.copy(default=data)
+            meeting_origin_vals.pop('id')
+            return meeting_origin.copy(default=meeting_origin_vals)
 
     @api.multi
     def action_detach_recurring_event(self):
+        """ Action used in the view to only update the virtual event we selected. It call the function to detach the virtual record and create a new from it.
+            It returns a form view in edit mode with the detached calendar event.
+        """
         meeting = self.detach_recurring_event()
         return {
             'type': 'ir.actions.act_window',
             'res_model': 'calendar.event',
             'view_mode': 'form',
             'res_id': meeting.id,
             'target': 'current',
-            'flags': {'form': {'action_buttons': True, 'options': {'mode': 'edit'}}}
+            'flags': {'form': {'action_buttons': True}, 'initial_mode': 'edit'},
+        }
+
+    @api.multi
+    def get_split_recurring_event(self):
+        """ Duplicate a calendar.event if we want to update all the virtual events from a fixed date.
+            In case of end type with count:
+                It will split the count between past events and remaining events. It will assign pas event count to original event
+                and the remaining count to the new event
+            If end type is a final_date:
+                It will set the final date of the original meeting to the selected event start date.
+                It will set the final date for the new event to the original final date.
+            Special case: If we selected the first event of the recurrence we will return the real id because there is no need to split between future and past events.
+            (update future events from the first event = update all events)
+            :return: A new record set containing the recurrence for future events.
+        """
+        meeting_origin, meeting_origin_vals = self.get_origin_meeting_and_schedule_fields()
+
+        reccurent_dates = meeting_origin._get_recurrent_date_by_event()
+        # If we want all the future event from the first event, that means we want all events and there is no need to split it.
+        real_start = pytz.UTC.localize(fields.Datetime.from_string(meeting_origin_vals['start'])).astimezone(pytz.UTC)
```

Comment:
Yep because you can't compare offset-naive and offset-aware datetimes.
- reccurent_dates has datetime with the time zone inside.
- pytz.UTC.localize add the tz-info in the datetime in order to compare it with the recurrent dates.